server_test.go

```diff
@@ -76,89 +79,90 @@ func runLsTestHelper(t *testing.T, result, expectedType, path string) {
 	// expected layout is:
 	// drwxr-xr-x   8 501      20            272 Aug  9 19:46 examples
 
-	// permissions (len 10, "drwxr-xr-x")
-	got := result[0:10]
-	if ok, err := regexp.MatchString("^"+expectedType+"[rwx-]{9}$", got); !ok {
-		t.Errorf("runLs(%#v, *FileInfo): permission field mismatch, expected dir, got: %#v, err: %#v", path, got, err)
+	t.Log(result)
+
+	sparce := strings.Split(result, " ")
+
+	var fields []string
+	for _, field := range sparce {
+		if field == "" {
+			continue
+		}
+
+		fields = append(fields, field)
 	}
```

Comment:
We do a little more robust of a text parsing here rather than strictly relying on things lining up. As any example, if a user had a username longer than 8 characters that could break the tests.